app/models/course/forum.rb

```diff
@@ -0,0 +1,21 @@
+class Course::Forum < ActiveRecord::Base
+  extend FriendlyId
+  friendly_id :slug_candidates, use: :slugged
+
+  belongs_to :course, inverse_of: :forums
+  has_many :topics, as: :commentable, class_name: Course::Discussion::Topic.name,
+                    dependent: :destroy, inverse_of: :commentable
+  has_many :subscriptions, class_name: Course::Forum::Subscription.name, dependent: :destroy,
+                           inverse_of: :forum
+
+  private
+
+  # Try building a slug based on the following fields in
+  # increasing order of specificity.
+  def slug_candidates
+    [
+      :name,
+      [:name, :course_id]
```

Comment:
The name of the forum should not have any enforcement of uniqueness, unlike filenames. Having a random string at the back of the slug is fine.